test/unit/transaction.test.js

```diff
@@ -23,528 +16,534 @@ const Segment = require('../../lib/transaction/trace/segment')
-      expect(trans.nameState.getName()).equal(null)
-    })
+    const TRACE_NAME = 'Custom/Test06'
+    const SLEEP_DURATION = 43
+    tt = new Transaction(agent)
 
-    it('should have no HTTP status code set (for hidden class)', function () {
-      expect(trans.statusCode).equal(null)
-    })
+    tt.measure(TRACE_NAME, null, SLEEP_DURATION)
+    tt.measure(TRACE_NAME, null, SLEEP_DURATION - 5)
 
-    it('should have no error attached (for hidden class)', function () {
-      expect(trans.error).equal(null)
-    })
+    const statistics = tt.metrics.getMetric(TRACE_NAME)
+    t.equal(
+      statistics.callCount,
+      2,
+      'should allow multiple overlapping metric measurements for same name'
+    )
+    t.ok(statistics.max > (SLEEP_DURATION - 1) / 1000, 'should measure at least 42 milliseconds')
 
-    it('should have no HTTP method / verb set (for hidden class)', function () {
-      expect(trans.verb).equal(null)
-    })
+    tt.end()
 
-    it('should not be ignored by default (for hidden class)', function () {
-      return expect(trans.ignore).false
-    })
+    tt = new Transaction(agent)
+    tt.measure('Custom/Test16', null, 65)
+    tt.end()
 
-    it('should not have a sampled state set', function () {
-      expect(trans.sampled).to.equal(null)
-    })
+    const metrics = tt.metrics.getMetric('Custom/Test16')
+    t.equal(metrics.total, 0.065, 'should allow manual setting of metric durations')
+
+    t.end()
   })
 
-  describe('with associated metrics', function () {
-    it('should manage its own independent of the agent', function () {
-      expect(trans.metrics).instanceOf(Metrics)
-      expect(trans.metrics).not.equal(getMetrics(agent))
-    })
+  t.test('when setting apdex for key transactions', (t) => {
+    txn._setApdex('Apdex/TestController/key', 1200, 667)
+    const metric = txn.metrics.getMetric('Apdex/TestController/key')
 
-    it('should have the same apdex threshold as the agent', function () {
-      expect(getMetrics(agent).apdexT).equal(trans.metrics.apdexT)
-    })
+    t.equal(metric.apdexT, 0.667, 'should set apdexT to the key transaction apdexT')
+    t.equal(metric.satisfying, 0, 'should not have satisfied')
+    t.equal(metric.tolerating, 1, 'should have been tolerated')
+    t.equal(metric.frustrating, 0, 'should not have frustrated')
 
-    it('should have the same metrics mapper as the agent', function () {
-      expect(agent.mapper).equal(trans.metrics.mapper)
-    })
+    txn._setApdex('Apdex/TestController/another', 1200)
+    const another = txn.metrics.getMetric('Apdex/TestController/another')
+    t.equal(another.apdexT, 0.1, 'should not require a key transaction')
+    t.end()
   })
+})
 
-  it('should know when it is not a web transaction', function () {
-    const tx = new Transaction(agent)
-    tx.type = Transaction.TYPES.BG
-    expect(tx.isWeb()).to.be.false
+tap.test('Transaction naming tests', (t) => {
+  t.autoend()
+  const agent = helper.loadMockedAgent({
```

Comment:
Previously, this test group used a new agent for each subtest. Is it important to preserve that?

Reply:
Okay, I've moved the tear-down/setup logic closer to each subtest to ensure the agent gets thrown out every time.